monitoring/rid_qualifier/flight_track_factory.py

```diff
@@ -0,0 +1,178 @@
+            return True
+        else: 
+            return False
+        
+    def generate_query_bboxes(self):
+        ''' For the differnet Remote ID checks: No, we need to generate three bounding boxes for the display provider, this method generates the 1 km diagonal length bounding box '''
+        # Get center of of the bounding box that is inputted into the generator
+        box = shapely.geometry.box(self.minx, self.miny, self.maxx, self.maxy)
+        center = box.centroid
+        # Transform to geographic co-ordinates to get areas
+        transformer = Transformer.from_crs("epsg:4326", "epsg:3857")
+        transformed_x,transformed_y = transformer.transform(center.x, center.y)
+        pt = Point(transformed_x,transformed_y)
+        # Now we have a point, we can buffer the point and create bounding boxes of the buffer to get the appropriate polygons, more than three boxes can be created, for the tests three will suffice.
+        box_diagonals = {1:{'length':150, 'name':'zoomed_in_detail'},2:{'length':380, 'name':"whole_flight_area"},3:{'length':3000, 'name':'too_large_query'}}
+
+        for box_id, box_diagonal in box_diagonals.items():
+            # Buffer the point with the appropriate length
+            buffer = pt.buffer(box_diagonal['length'])                   
+            buffer_bounds = buffer.bounds
+            buffer_bounds_polygon  =  shapely.geometry.box(buffer_bounds[0], buffer_bounds[1], buffer_bounds[2], buffer_bounds[3])
+            buffer_points =  zip(buffer_bounds_polygon.exterior.coords.xy[0], buffer_bounds_polygon.exterior.coords.xy[1])
+            proj_buffer_points = []
+            # reproject back to ESPG 4326
+            transformer2 = Transformer.from_crs("epsg:3857", "epsg:4326")
+            for point in buffer_points:
+                x = point[0]
+                y = point[1]
+                x, y = transformer2.transform(x, y)
+                proj_buffer_points.append((x, y))
+            
+            buffered_box = Polygon(proj_buffer_points)
+            # Get the bounds of the buffered box, this is the one that will will be fed to the remote ID display provider to query
+            self.query_bboxes.append({'name':box_diagonals[box_id]['name'], 'shape': buffered_box})
+            
+
+    def generate_flight_grid(self):
+        ''' Generate a series of boxes within the bounding box to have areas for different flights '''
+        # Compute the box where the flights will be created. For a the sample bounds given, over Bern, Switzerland, a division by 2 produces a cell_size of 0.0025212764739985793, a division of 3 is 0.0016808509826657196 and division by 4 0.0012606382369992897. As the cell size goes smaller more number of flights can be accomodated within the grid. For the study area bounds we build a 3x2 box for six flights by creating 3 column 2 row grid. 
+
+        cell_size_x = (self.maxx - self.minx)/(3) # create three columns
```

Comment:
To avoid accidental inconsistencies when changing values later and to increase readability, let's define constants:

```python
N_COLS = 3

[...]

cell_size_x = (self.maxx - self.minx) / N_COLS

[...]

for u0 in range(0, N_COLS):
```

Reply:
Done! Also for rows. Thanks!